kedro-datasets/kedro_datasets/_io.py

```diff
@@ -0,0 +1,12 @@
+"""
+Adapter for kedro.io.core for backwards compatibility.
+"""
+
+try:
+    # kedro 0.18.11 onwards
```

Comment:
Is it better to just test `kedro.__version__`. Believe that's what we do in other places.

Reply:
Maybe I'm thinking of viz: https://github.com/kedro-org/kedro-viz/blob/8e4ab14671d1ccc800a6425b1381cf5d03845cca/package/kedro_viz/data_access/repositories/catalog.py#L59

Reply:
This mirrors what's done for newer Python features. Couldn't find an authoritative source for it, but essentially what https://stackoverflow.com/a/59734959/554319 shows:

```
try:
    from importlib import metadata
except ImportError: # for Python<3.8
    import importlib_metadata as metadata
```

Reply:
Sounds good to me!